### ISSUE TITLE
Clarity around target-action in the documentation

### ISSUE THREAD

**Issue by cbandy** [pgbackrest/pgbackrest#987]
The [`restore --target-action` documentation](https://pgbackrest.org/command.html#command-restore/category-command/option-target-action) states:

> This option is effective when `hot_standby=on` is configured in `postgresql.conf`, otherwise the cluster will be promoted when the target is reached or there is no more WAL in the archive.

However, I don't believe this is accurate. In recent versions of PostgreSQL, [1][2] `recovery_target_action` is considered anytime one of `recovery_target`, `recovery_target_name`, `recovery_target_time`, `recovery_target_xid`, or `recovery_target_lsn` is set. When `hot_standby` is enabled and the target is reached,
- `promote` ends recovery, picks a new timeline, and accepts connections.
- `pause` leaves the cluster in recovery and accepts connections.
- `shutdown` stops the cluster.

When `hot_standby` is disabled, the only difference is that `pause` acts like `shutdown`.

[1]\: https://www.postgresql.org/docs/12/runtime-config-wal.html
[2]\: https://www.postgresql.org/docs/11/recovery-target-settings.html

On PostgreSQL 12, I have observed that when the recovery target is reached:
- `hot_standby=on recovery_target_time={timestamp}` accepts connections and `pg_is_in_recovery() is true`. The default is to pause, as documented.

- `hot_standby=off recovery_target_time={timestamp}` stops the cluster. The cluster is not promoted, seemingly contrary to this documentation.

- `hot_standby=off recovery_target_time={timestamp} recovery_target_action=promote` ends recovery and accepts connections. The option is effective, seemingly contrary to this documentation.

**Comment by cbandy** [comment]
Instead of that sentence, I want to recommend (something like) the following:

> The following actions are supported:
> - `pause` - pause when recovery target is reached on a hot standby. (**PostgreSQL** >= 9.1)
> …
>
> ***NOTE:*** When `hot_standby=off` in `postgresql.conf`, the behavior of `pause` depends on the version of PostgreSQL.
> - PostgreSQL >= 9.5 will shutdown when the recovery target is reached.
> - PostgreSQL < 9.5 will promote and switch timeline when recovery target is reached.

But at the same time, neither of those sounds at all like "pause." Thoughts?

The behavior of PostgreSQL < 9.5 is based on my reading of the docs; I did not test it. Is it behavior that you expect as well?

**Comment by cbandy** [comment]
I discovered another wrinkle while researching this topic. Before PostgreSQL 12, the "pause-means-shutdown" behavior requires an explicit `recovery_target_action=pause` in `recovery.conf`. Notice `recoveryTargetActionSet` below:

https://github.com/postgres/postgres/blob/REL_11_7/src/backend/access/transam/xlog.c#L5551-L5554
```c
	if (recoveryTargetAction == RECOVERY_TARGET_ACTION_PAUSE &&
		recoveryTargetActionSet &&
		!EnableHotStandby)
		recoveryTargetAction = RECOVERY_TARGET_ACTION_SHUTDOWN;
```

**Comment by dwsteele** [comment]
Yes, there have been a lot of changes in this area over the years and it can be hard to keep the documentation up to date. It was mostly written back in the days of 9.5.

> But at the same time, neither of those sounds at all like "pause." Thoughts?

When `hot-standby=off` pause is ignored, so we should probably stick to that.

> The behavior of PostgreSQL < 9.5 is based on my reading of the docs; I did not test it. Is it behavior that you expect as well?

Not sure, I'd need to spend some time having a look. We do many of these scenarios from PG 8.3 to 12 so I'm reasonably confident that things work as expected in most cases. However, what the tests expect and what's in the documentation are not necessarily the same thing, as you have noted.

> I discovered another wrinkle while researching this topic. Before PostgreSQL 12, the "pause-means-shutdown" behavior requires an explicit recovery_target_action=pause in recovery.conf

I reported this last year but didn't follow up and it seems nobody else did either:

https://www.postgresql.org/message-id/flat/c39ffa6a-f44f-bab4-853b-66157ca8d90f%40pgmasters.net

So, seems like a good idea to patch the documentation prior to PG12 to reflect this.

**Comment by cbandy** [comment]
The PostgreSQL releases since 12 have behaved consistently, so I think the following covers all the issues: 

1. The default value for `hot_standby` changed in PostgreSQL 10. Mention that without describing what is or isn't in the `postgresql.conf` file.
2. The default (unconfigured) behavior of `recovery_target_action` changed in PostgreSQL 12.

> Action to take when recovery target is reached.
>
> When `hot_standby=on`, the default since **PostgreSQL** 10, this option consistently controls what the cluster does when the target is reached or there is no more WAL in the archive.
>
> When `hot_standby=off` in **PostgreSQL** >= 12, `pause` acts like `shutdown`.
> When `hot_standby=off` in **PostgreSQL** < 12, `pause` acts like `promote`.
>
> The following actions are supported:
> …

**Comment by dwsteele** [comment]
This appears to be fixed by PR #1960.
